Dockerfile

```diff
@@ -1,34 +1,47 @@
-FROM node:12-stretch
-LABEL description="Debian image to build nativefier apps"
+FROM node:12-alpine
+LABEL description="Alpine image to build Nativefier apps"
 
-# Get wine32, not 64, to work around binary incompatibility with rcedit.
-# https://github.com/nativefier/nativefier/issues/375#issuecomment-304247033
-# Forced us to use Debian rather than Alpine, which doesn't do multiarch.
-RUN dpkg --add-architecture i386
 
 # Install dependencies
-RUN apt-get update \
-    && apt-get --yes install wine32 imagemagick \
-    && apt-get clean && rm -rf /var/lib/apt/lists/*
+RUN apk update \
+    && apk add bash wine imagemagick dos2unix \
+    && rm -rf /var/cache/apk/*
+
+WORKDIR /nativefier
 
 # Add sources
-COPY . /nativefier
+COPY . .
+
+# Fix line endings that may have gotten mangled in Windows
+RUN find ./icon-scripts ./src ./app -type f -print0 | xargs -0 dos2unix
 
 # Build nativefier and link globally
 WORKDIR /nativefier/app
 RUN npm install
 WORKDIR /nativefier
-RUN npm install && npm run build && npm link
+
+# Install (which will also install in ./app, and build, thanks to our `prepare` npm hook)
+# Also, running tests, to ensure we don't Docker build & publish broken stuff
+RUN npm install && npm run build && npm test && npm link
```

Comment:
**Last remaining question**: In an earlier comment I was suggesting that `npm run build` is unnecessary, since already run through `prepare` after `install`. Was I wrong, or did my message get lost during the review? If not,

```suggestion
RUN npm install && npm test && npm link
```

Reply:
It did not get lost. I apologize for not communicating, but the npm install inside the docker image did not seem to execute the prepare:

```#13 [ 9/12] RUN npm install && npm test && npm link
#13 13.86 npm WARN deprecated cheerio-select-tmp@0.1.1: Use cheerio-select instead
#13 22.44 npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
#13 22.45 npm WARN deprecated request-promise-native@1.0.9: request-promise-native has been deprecated because it extends the now deprecated request package, see https://github.com/request/request/issues/3142
#13 25.06 npm WARN deprecated har-validator@5.1.5: this library is no longer supported
#13 27.94 npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
#13 27.98 npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
#13 60.33
#13 60.33 > core-js@3.9.1 postinstall /nativefier/node_modules/core-js
#13 60.33 > node -e "try{require('./postinstall')}catch(e){}"
#13 60.33
#13 60.46 Thank you for using core-js ( https://github.com/zloirock/core-js ) for polyfilling JavaScript standard library!
#13 60.46
#13 60.46 The project needs your help! Please consider supporting of core-js on Open Collective or Patreon:
#13 60.46 > https://opencollective.com/core-js
#13 60.46 > https://www.patreon.com/zloirock
#13 60.46
#13 60.46 Also, the author of core-js ( https://github.com/zloirock ) is looking for a good job -)
#13 60.46
#13 61.65 npm WARN lifecycle nativefier@42.3.0~prepare: cannot run in wd nativefier@42.3.0 cd app && npm install && cd .. && npm run build (wd=/nativefier)
#13 62.53 npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^2.1.2 (node_modules/jest-haste-map/node_modules/fsevents):
#13 62.53 npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
#13 62.53
#13 62.57 added 858 packages from 522 contributors and audited 861 packages in 60.667s
#13 63.63
#13 63.63 76 packages are looking for funding
#13 63.63   run `npm fund` for details
#13 63.63
#13 63.63 found 0 vulnerabilities
#13 63.63
#13 64.17
#13 64.17 > nativefier@42.3.0 test /nativefier
#13 64.17 > jest --testRegex '[-.]test\.js$'
#13 64.17
#13 64.63 ● Validation Error:
#13 64.63
#13 64.63   Module ./lib/jestSetupFiles in the setupFiles option was not found.
#13 64.63          <rootDir> is: /nativefier
#13 64.63
#13 64.63   Configuration Documentation:
#13 64.63   https://jestjs.io/docs/configuration.html
#13 64.63
#13 64.64 npm ERR! Test failed.  See above for more details.
#13 ERROR: executor failed running [/bin/sh -c npm install && npm test && npm link]: exit code: 1
------
 > [ 9/12] RUN npm install && npm test && npm link:
------
executor failed running [/bin/sh -c npm install && npm test && npm link]: exit code: 1
```

Reply:
Digging in a little bit, it appears that for safety, `prepare` won't run as root:

https://github.com/npm/npm/issues/17346

Reply:
> Digging in a little bit, it appears that for safety, `prepare` won't run as root: [npm/npm#17346](https://github.com/npm/npm/issues/17346)

@TheCleric ooooh, okay, nice find. Adding a lil' comment to document it's necessary, and a link to the npm issue.

Reply:
I'm seeing if I have a workaround if you want, or we can merge it as is.

Reply:
Nope, didn't work. 😆

Reply:
Nah it's all good.

But wait, how does thing even work, then?! (No, I'm not super familiar with our Docker build 😄). If `prepare` doesn't run on `install`, it means that the `cd app && npm install && cd ..` bit of `prepare` doesn't run either, so the app will generate, but will be broken at runtime as missing dependencies (for the app, not the build time CLI builder). If it used to be the case, Docker users (and I know we have a few) would have complained about it, so I understand that the build behaved differently before your PR? However, that doesn't make sense, the position of the `USER` statement before the npm install didn't budge with your PR, so we should have been making broken Docker builds before your PR.

I don't understand what's going on. Do you? Do the app generated by Docker work?

Reply:
They work because we're manually doing those steps:

```
# Build nativefier and link globally
WORKDIR /nativefier/app
RUN npm install
```

Reply:
Aaaaaaaaaalright, makes sense, had missed those.